spot-contracts/contracts/FeePolicy.sol

```diff
@@ -0,0 +1,259 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import { IFeePolicy, IERC20Upgradeable } from "./_interfaces/IFeePolicy.sol";
+import { IPerpetualTranche, IBondIssuer } from "./_interfaces/IPerpetualTranche.sol";
+import { IVault } from "./_interfaces/IVault.sol";
+
+import { MathUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/math/MathUpgradeable.sol";
+import { SafeCastUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/math/SafeCastUpgradeable.sol";
+import { AddressUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Sigmoid } from "./_utils/Sigmoid.sol";
+
+/**
+ *  @title FeePolicy
+ *
+ *  @notice This contract determines fees for interacting with the perp and vault systems.
+ *
+ *          The fee policy attempts to balance the demand for holding perp tokens with
+ *          the demand for holding vault tokens; such that the total collateral in the vault
+ *          supports rolling over all mature collateral backing perps.
+ *
+ *          Fees are computed based on the following variables.
+ *              - `currentVaultTVL`  : Total value of collateral in the rollover vault at a given time.
+ *              - `targetVaultTVL`   : The minimum value of collateral that needs to be in the vault
```

Comment:
Might be worth talking about naming, since I've had to reference this a bunch to times to keep straight.

The targetTVL here is not actually "the target" as far as behavior goes. The target the system tries to meet is targetSubscriptionRatio * targetTVL.

What about "bondEquilibriumTVL" being what matches the bond ratios
Then target is what's left after the safety buffer is applied?